### ISSUE TITLE
Bug in DoHeatmap legend

### ISSUE THREAD

**Issue by bvaldebenitom** [satijalab/seurat#5048]
Hi all!

Thanks for Seurat. It's an amazing tool. There seems to be a bug with the DoHeatmap function.

Output of DoHeatmap shows random characters in the legend
![imagen](https://user-images.githubusercontent.com/31257257/132034016-24d97482-b3cd-4c2e-befb-a54bb53f7358.png)

This is the smallest simplified code that reproduces the issue. But I get the same problem when doing all the standard processing steps.

```r
library(Seurat)
data(pbmc_small)
pbmc_small <- ScaleData(pbmc_small, features = rownames(pbmc_small))
DoHeatmap(pbmc_small,features=c("TYROBP"))
```

sessionInfo() output
```
R version 4.0.4 (2021-02-15)
Platform: x86_64-apple-darwin18.7.0 (64-bit)
Running under: macOS Mojave 10.14.6

Matrix products: default
BLAS:   /usr/local/lib/R/lib/libRblas.dylib
LAPACK: /usr/local/lib/R/lib/libRlapack.dylib

locale:
[1] C/UTF-8/C/C/C/C

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] SeuratObject_4.0.2 Seurat_4.0.3      

loaded via a namespace (and not attached):
  [1] nlme_3.1-152          matrixStats_0.59.0    spatstat.sparse_2.0-0
  [4] RcppAnnoy_0.0.18      RColorBrewer_1.1-2    httr_1.4.2           
  [7] sctransform_0.3.2     tools_4.0.4           utf8_1.2.2           
 [10] R6_2.5.0              irlba_2.3.3           rpart_4.1-15         
 [13] KernSmooth_2.23-20    uwot_0.1.10           mgcv_1.8-36          
 [16] DBI_1.1.1             lazyeval_0.2.2        colorspace_2.0-2     
 [19] tidyselect_1.1.1      gridExtra_2.3         compiler_4.0.4       
 [22] plotly_4.9.4          labeling_0.4.2        scales_1.1.1         
 [25] lmtest_0.9-38         spatstat.data_2.1-0   ggridges_0.5.3       
 [28] pbapply_1.4-3         goftest_1.2-2         stringr_1.4.0        
 [31] digest_0.6.27         spatstat.utils_2.1-0  pkgconfig_2.0.3      
 [34] htmltools_0.5.1.1     parallelly_1.26.0     fastmap_1.1.0        
 [37] htmlwidgets_1.5.3     rlang_0.4.11          shiny_1.6.0          
 [40] farver_2.1.0          generics_0.1.0        zoo_1.8-9            
 [43] jsonlite_1.7.2        ica_1.0-2             dplyr_1.0.7          
 [46] magrittr_2.0.1        patchwork_1.1.1       Matrix_1.3-4         
 [49] Rcpp_1.0.7            munsell_0.5.0         fansi_0.5.0          
 [52] abind_1.4-5           reticulate_1.20       lifecycle_1.0.0      
 [55] stringi_1.6.2         MASS_7.3-54           Rtsne_0.15           
 [58] plyr_1.8.6            grid_4.0.4            parallel_4.0.4       
 [61] listenv_0.8.0         promises_1.2.0.1      ggrepel_0.9.1        
 [64] crayon_1.4.1          miniUI_0.1.1.1        deldir_0.2-10        
 [67] lattice_0.20-44       cowplot_1.1.1         splines_4.0.4        
 [70] tensor_1.5            pillar_1.6.2          igraph_1.2.6.9118    
 [73] spatstat.geom_2.1-0   future.apply_1.7.0    reshape2_1.4.4       
 [76] codetools_0.2-18      leiden_0.3.8          glue_1.4.2           
 [79] data.table_1.14.0     png_0.1-7             vctrs_0.3.8          
 [82] httpuv_1.6.1          gtable_0.3.0          RANN_2.6.1           
 [85] purrr_0.3.4           spatstat.core_2.1-2   polyclip_1.10-0      
 [88] tidyr_1.1.3           scattermore_0.7       future_1.21.0        
 [91] assertthat_0.2.1      ggplot2_3.3.5         mime_0.10            
 [94] xtable_1.8-4          later_1.2.0           survival_3.2-11      
 [97] viridisLite_0.4.0     tibble_3.1.3          cluster_2.1.2        
[100] globals_0.14.0        fitdistrplus_1.1-5    ellipsis_0.3.2       
[103] ROCR_1.0-11     
```

**Comment by saketkc** [comment]
Thanks for reporting. This bug was fixed in [4.0.4](https://satijalab.org/seurat/news/index.html#seurat-4-0-4-2020-08-19-2021-08-20). You can update Seurat and report back if you still notice issues.

**Comment by bvaldebenitom** [comment]
Thanks for the quick reply. Updated it and the bug is gone!
